servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/HttpResponseDecoder.java

```diff
@@ -140,12 +150,18 @@ protected boolean isContentAlwaysEmpty(final HttpResponseMetaData msg) {
 
     private static int nettyBufferToStatusCode(final ByteBuf buffer, final int start, final int length) {
         if (length != 3) {
-            splitInitialLineError();
+            throw new DecoderException("Invalid start-line: HTTP status-code must contain only 3 digits, found: " +
+                    buffer.toString(start, length, US_ASCII));
         }
 
         final int medium = buffer.getUnsignedMedium(start);
-        return toDecimal((medium & 0xff0000) >> 16) * 100 +
-                toDecimal((medium & 0xff00) >> 8) * 10 +
-                toDecimal(medium & 0xff);
+        try {
+            return toDecimal((medium & 0xff0000) >> 16) * 100 +
+                    toDecimal((medium & 0xff00) >> 8) * 10 +
+                    toDecimal(medium & 0xff);
+        } catch (IllegalCharacterException cause) {
+            throw new DecoderException("Invalid start-line: HTTP status-code must contain only 3 digits, found: " +
```

Comment:
we should consider moving the exceptional cases that have to create exceptions and build strings separate methods to allow the JVM to inline more easily for less frequent code paths.